backend/Application/Aggregates/Contract/Jobs/InitialContractEventDeserializationFieldsCatchUpJob.cs

```diff
@@ -69,19 +72,33 @@ await Policies.GetTransientPolicy(GetUniqueIdentifier(), _logger, _contractAggre
                 await using var moduleReadonlyRepository = new ModuleReadonlyRepository(context);
 
                 var contractEvents = await context.ContractEvents
+                    .AsNoTracking()
                     .OrderBy(ce => ce.BlockHeight)
                     .ThenBy(ce => ce.TransactionIndex)
                     .ThenBy(ce => ce.EventIndex)
                     .Skip(skip)
                     .Take(take)
                     .ToListAsync(token);
 
+                var eventUpdates = new List<Update>();
                 foreach (var contractEvent in contractEvents
-                             .Where(contractEvent => !contractEvent.IsHexadecimalFieldsParsed()))
+                             .Where(contractEvent => !IsHexadecimalFieldsParsed(contractEvent)))
                 {
                     try
                     {
-                        await contractEvent.ParseEvent(contractRepository, moduleReadonlyRepository);
+                        var transactionResultEvent = await TryParseEvent(contractEvent, contractRepository, moduleReadonlyRepository);
+                        if (transactionResultEvent == null)
+                        {
+                            continue;
+                        }
+                        eventUpdates.Add(new Update(
```

Comment:
I would move this into the conditional statement and check for `transactionResultEvent != null`.